t/data_migration_pipeline/main_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/passert"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/ptest"
+)
+
+func TestEmitResult(t *testing.T) {
```

Comment:
Could we automate the setup of the spanner test db using `pkg/containertest`?

Reply:
Approving as I don't think its a blocker, but would be nice if we could test the template in CI that way.

Reply:
Sure we could do that. By achieving an integration test, we'll do
- Docker run spanner emulator
- Docker run tests with integration env set up, a example: https://github.com/apache/beam/blob/master/sdks/go/test/integration/io/spannerio/test_helpers.go
That might be too much for a boilerplate template, let's add that later if an integration tests is needed for this pipeline, or we can trust beam io package and make this code piece light weight without integration testing.